e2e-tests/development-runtime/cypress/integration/hot-reloading/non-js-file.js

```diff
@@ -3,31 +3,63 @@ const TEST_ID = `sub-title`
 const message = `This is a sub-title`
 
 describe(`hot reloading non-js file`, () => {
-  beforeEach(() => {
-    cy.exec(
-      `npm run update -- --file content/2018-12-14-hello-world.md --replacements "${message}:%${TEMPLATE}%" --exact`
-    )
-    cy.wait(1000)
-
-    cy.visit(`/2018-12-14-hello-world/`).waitForRouteChange()
-
-    cy.wait(1000)
+  describe(`markdown`, () => {
+    beforeEach(() => {
+      cy.exec(
+        `npm run update -- --file content/2018-12-14-hello-world.md --replacements "${message}:%${TEMPLATE}%" --exact`
+      )
+      cy.wait(1000)
+  
+      cy.visit(`/2018-12-14-hello-world/`).waitForRouteChange()
+  
+      cy.wait(1000)
+    })
+  
+    it(`displays placeholder content on launch`, () => {
+      cy.getTestElement(TEST_ID).invoke(`text`).should(`contain`, TEMPLATE)
+    })
+  
+    it(`hot reloads with new content`, () => {
+      cy.getTestElement(TEST_ID).invoke(`text`).should(`contain`, TEMPLATE)
+  
+      cy.exec(
+        `npm run update -- --file content/2018-12-14-hello-world.md --replacements "${TEMPLATE}:${message}"`
+      )
+  
+      // wait for socket.io to update
+      cy.wait(5000)
+  
+      cy.getTestElement(TEST_ID).invoke(`text`).should(`eq`, message)
+    })
   })
 
-  it(`displays placeholder content on launch`, () => {
-    cy.getTestElement(TEST_ID).invoke(`text`).should(`contain`, TEMPLATE)
-  })
+  describe(`image`, () => {
+    beforeEach(() => {
+      cy.exec(
+        `npm run update -- --file src/images/image.png --copy "src/images/original.png"`
+      )
+      cy.wait(1000)
+  
+      cy.visit(`/md-image/`).waitForRouteChange()
+      cy.wait(1000)
+    })
 
-  it(`hot reloads with new content`, () => {
-    cy.getTestElement(TEST_ID).invoke(`text`).should(`contain`, TEMPLATE)
+    it(`displays original content on launch`, () => {
+      cy.matchImageSnapshot(`non-js-file--image-original`)
```

Comment:
When working with image snapshots we often are forcing particular viewport ( https://docs.cypress.io/api/commands/viewport ). This is only to make sure some default won't result in different screenshot "size" on different machines / environments.

Using `cy.viewport` might fix https://dashboard.cypress.io/projects/s3j3qj/runs/18415/test-results/223cff5d-8035-472f-b3f8-2175f094d30d 

```
Image size (1000x734) different than saved snapshot size (1000x733).

See diff for details: /tmp/tmp.klQmhFMKrw/e2e-tests/development-runtime/diff_output/hot-reloading/non-js-file.js/non-js-file--image-original.diff.png
```

Reply:
Good idea. I made the test grab the individual image elements from the dom instead.